crates/sui-core/src/state_accumulator.rs

```diff
@@ -61,29 +61,18 @@ impl StateAccumulator {
         );
 
         // get all modified_at_versions for the fx
-        let modified_at_version_keys = effects
-            .iter()
-            .flat_map(|fx| {
-                fx.modified_at_versions()
-                    .iter()
-                    .map(|(id, seq_num)| ObjectKey(*id, *seq_num))
-                    .collect::<Vec<ObjectKey>>()
-            })
-            .collect::<HashSet<ObjectKey>>();
-
-        let modified_at_digests = self
-            .authority_store
-            .perpetual_tables
-            .parent_sync
-            .iter()
-            .filter_map(|(tup, _tx_digest)| {
-                if modified_at_version_keys.contains(&ObjectKey(tup.0, tup.1)) {
-                    Some(tup.2)
-                } else {
-                    None
-                }
-            })
-            .collect::<Vec<ObjectDigest>>();
+        let modified_at_version_keys = effects.iter().flat_map(|fx| {
+            fx.modified_at_versions()
+                .iter()
+                .map(|(id, seq_num)| ObjectKey(*id, *seq_num))
+                .collect::<Vec<ObjectKey>>()
+        });
+        let mut modified_at_digests = vec![];
+        for key in modified_at_version_keys {
+            if let Some(object) = self.authority_store.get_object_by_key(&key.0, key.1)? {
```

Comment:
not fully familiar with state accumulator design, is there a concern that pruner might remove object version before accumulator reads it?

Reply:
accumulator only processes checkpointed effects, so it has no interaction with pruner

Reply:
just curious, does accumulator block checkpoint executor somehow  ?
I changed iterator over parent_sync table to fetching object by (object_id, version). 
I wonder if that's ok. Should be fine now cause pruner's default setting is to keep one extra epoch for data.
But I wonder if any conflict possible with aggressive pruner setting (i.e. pruner was faster and removed entry that accumulator still needs)

Reply:
discussed offline. Not an issue for now because if keep last epoch of data. But we need to ensure that pruner respects  checkpoint watermark for aggressive setup